src/openzeppelin/token/erc1155/presets/ERC1155MintableBurnable.cairo

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: MIT
+# OpenZeppelin Contracts for Cairo v0.2.1 (token/erc1155/presets/ERC1155MintableBurnable.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.math import assert_not_zero
+
+from openzeppelin.access.ownable.library import Ownable
+from openzeppelin.token.erc1155.library import ERC1155
+from openzeppelin.introspection.erc165.library import ERC165
+
+#
+# Constructor
+#
+
+@constructor
+func constructor{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(uri: felt, owner: felt):
+    ERC1155.initializer(uri)
+    Ownable.initializer(owner)
+    return ()
+end
+
+#
+# Getters
+#
+
+@view
+func supportsInterface{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(interfaceId: felt) -> (success: felt):
+    return ERC165.supports_interface(interfaceId)
+end
+
+@view
+func uri{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(id: Uint256) -> (uri: felt):
+    return ERC1155.uri(id)
+end
+
+@view
+func balanceOf{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(account: felt, id: Uint256) -> (balance: Uint256):
+    return ERC1155.balance_of(account, id)
+end
+
+@view
+func balanceOfBatch{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        accounts_len: felt,
+        accounts: felt*,
+        ids_len: felt,
+        ids: Uint256*
+    ) -> (balances_len: felt, balances: Uint256*):
+    let (balances_len, balances) =  ERC1155.balance_of_batch(accounts_len, accounts, ids_len, ids)
+    return (balances_len, balances)
+end
+
+@view
+func isApprovedForAll{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(account: felt, operator: felt) -> (isApproved: felt):
+    let (is_approved) = ERC1155.is_approved_for_all(account, operator)
+    return (is_approved)
+end
+
+@view
+func owner{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }() -> (owner: felt):
+    let (owner: felt) = Ownable.owner()
+    return (owner)
+end
+
+#
+# Externals
+#
+
+@external
+func setURI{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(uri: felt):
+    Ownable.assert_only_owner()
+    ERC1155._set_uri(uri)
+    return ()
+end
```

Comment:
Hey, first thanks for this amazing job @dewi-tim, it will be very useful for the community.

I wanted to warn you about a little something, in the proposition you manage the URI as a felt, which will cap the string length to the short string max length (34 characters if I remember well).
Most of time it will be longer I think, therefore managing it with a pair `(uri_len : felt, uri : felt*)` will definitely help to workaround this limitation.

What do you think about it?

Reply:
Thanks! See the comment [here](https://github.com/OpenZeppelin/cairo-contracts/pull/290#discussion_r949660263) for the reasons why. Keeping it as a felt (and changing the event) to be consistent with the rest of the repo, but yeah, short strings are a little short.